packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js

```diff
@@ -87,26 +124,111 @@ export default class QuanticSearchBoxSuggestionsList extends LightningElement {
     }
   }
 
-  get suggestionsToRender() {
-    return this.suggestions?.map((s, i) => ({
-      ...s,
-      isSelected: this.selectionIndex === i,
-    }));
+  get allOptions() {
```

Comment:
For your code snippet suggestion I think it will not work properly because the the indexes will be duplicated we will have more than one option with index 0 for example, one for the first recent query and one for the first suggestion.

Reply:
Yea my bad, then you need to calculate the `index` at the end 😂 

```
return [
  ...recentQueriesOptions,
  ...querySuggestions
].map((suggestion, index) => ({...suggestion, key: index}).slice(0, this.maxNumberOfSuggestions + 1);
```

Reply:
We have to simplify this, it's really hard to read.